onmt/modules/Transformer.py

```diff
@@ -55,60 +35,45 @@ def __init__(self, hidden_size, dropout, padding_idx,
         self.feed_forward = PositionwiseFeedForward(hidden_size,
                                                     d_inner,
                                                     dropout)
-        self.padding_idx = padding_idx
-
-    def forward(self, input, words):
-        # CHECKS
-        n_batch, s_len, _ = input.size()
-        n_batch_, s_len_ = words.size()
-        aeq(n_batch, n_batch_)
-        aeq(s_len, s_len_)
-        # END CHECKS
 
-        mask = get_attn_padding_mask(words, words, self.padding_idx)
+    def forward(self, input, mask):
         mid, _ = self.self_attn(input, input, input, mask=mask)
         out = self.feed_forward(mid)
         return out
 
 
 class TransformerDecoder(nn.Module):
     """
-    The Transformer Decoder from AIAYN
+    The Transformer Decoder from paper "Attetion is all you need".
     """
-    def __init__(self, hidden_size, opt, pad,
+    def __init__(self, hidden_size, dropout,
```

Comment:
Can you update these argument names. 

hidden_size -> size
d_inner -> hidden_size 
n_head -> head_count

Also can you add an assert that  `size` be divisible by `head_count` 

Thanks.

Reply:
Sure.